proposals/dictionary-expressions.md

```diff
@@ -165,47 +165,52 @@ From [Open Question 3](#open-question-3), we will support *dictionary elements*
->   - ```diff
->     + If the collection expression has any elements and the type 
->     + has an iteration type of some `KeyValuePair<TKey, TValue>`
->     + and the type has applicable indexer that can be invoked with
->     + a single argument of the `TKey` type, and a value of the `TValue`
->     + type, and the indexer is accessible at the location of the
->     + collection expression.
->     ```
-> - An interface type:
->   - `System.Collections.Generic.IEnumerable<T>`
->   - `System.Collections.Generic.IReadOnlyCollection<T>`
->   - `System.Collections.Generic.IReadOnlyList<T>`
->   - `System.Collections.Generic.ICollection<T>`
->   - `System.Collections.Generic.IList<T>`
->   - ```diff
->     + System.Collections.Generic.IDictionary<TKey, TValue>
->     + System.Collections.Generic.IReadOnlyDictionary<TKey, TValue>
->     ```
+*Collection expression conversions* are updated to include conversions to *dictionary types*.
+
+An implicit *collection expression conversion* exists from a collection expression to the following *dictionary types*:
+* A *dictionary type* with an appropriate *[create method](#create-methods)*.
+* A *struct* or *class* *dictionary type* that implements `System.Collections.IEnumerable` where:
+  * The *element type* is determined from a `GetEnumerator` instance method or enumerable interface.
+  * The *type* has an *[applicable](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#11642-applicable-function-member)* constructor that can be invoked with no arguments, and the constructor is accessible at the location of the collection expression.
+  * The *indexer* has a setter that is as accessible as the declaring type.
+* An *interface type*:
+  * `System.Collections.Generic.IDictionary<TKey, TValue>`
+  * `System.Collections.Generic.IReadOnlyDictionary<TKey, TValue>`
+
+*Collection expression conversions* require implicit conversions for each element.
+The element conversion rules are now differentiated based on whether the *element type* of the target type is `KeyValuePair<,>`.
+(That includes target types other than dictionary types.)
+
+If the *element type* is a type *other than* `KeyValuePair<,>`, the rules are *unchanged* from *language version 12* other than **clarifications**:
+
+> An implicit *collection expression conversion* exists from a collection expression to a *type* with *element type* `T` **where `T` is not `KeyValuePair<,>` and** where for each *element* `Eᵢ` in the collection expression:
+> * If `Eᵢ` is an *expression element*, there is an implicit conversion from `Eᵢ` to `T`.
+> * If `Eᵢ` is a *spread element* `..Sᵢ`, there is an implicit conversion from the *iteration type* of `Sᵢ` to `T`.
+> * **Otherwise there is *no implicit conversion* from the collection expression to the target type.**
+
+If the *element type* is `KeyValuePair<,>` the rules are *modified* for *language version 13*:
```

Comment:
we might want to be explicit that these changes apply do non-dictionary collection types as well.

Reply:
Moved the explicit statement here.